src/components/Rover.tsx

```diff
@@ -1,25 +1,49 @@
 import { useParams, Redirect } from "react-router-dom";
 import React from "react";
+import { Card } from "./Card";
+import { rovers } from "./RoverData";
+import "../styles/RoverPage.scss";
 
 interface RoverParams {
     rover: string;
 }
 
 export function Rover() {
-    let { rover } = useParams<RoverParams>();
+    const { rover } = useParams<RoverParams>();
     const regexMatch = /(opportunity)|(spirit)|(curiosity)/i;
 
-    if(!rover.match(regexMatch)){
+    if (!rover.match(regexMatch)) {
         return (
             <Redirect to="/" />
         );
     }
 
-    rover = rover.substr(0, 1).toUpperCase() + rover.substr(1).toLowerCase();
-
     return (
-        <div>
-            <h2 className='current-page'>Hello {rover}</h2>
+        <div className = "rover-page">
+            <div data-testid={rover}>{findRover(rover)}</div>
+            <div>Big Photo</div>
+            <div>Gallery</div>
         </div>
     );
 }
+
+function findRover(rover: string) {
+    let data;
+    switch (rover) {
+    case "spirit":
+        data = rovers[0];
+        break;
+    case "opportunity":
+        data = rovers[1];
+        break;
+    case "curiosity":
+        data = rovers[2];
+        break;
+    }
+
+    if (data) {
```

Comment:
Is there a reason why we need this? AFAICT (As far as I can tell) if we do catch all the case insensitives cases above, then data should always match?

Reply:
Done